### PR TITLE
update to 0.5

### PR DESCRIPTION
**Pull request by r-chris**
Adding compatibility for julia 0.5

### CODE REVIEW

File: bin/jdp [hunk 1]
```diff
@@ -60,9 +59,9 @@ DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
 
 # install packages if not installled yet
```
Comment (r-chris) [inline bin/jdp:61]: installled -> installed

**Review by r-chris — COMMENTED** [review]
Nothing major - good example for what needs to change when switching to 0.5 - we should actually write a script for this stuff :+1:

File: src/DeclarativePackages.jl [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+    @eval is_apple()   = $(KERNEL == :Darwin)
+    @eval is_linux()   = $(KERNEL == :Linux)
+    @eval is_bsd()     = $(KERNEL in (:FreeBSD, :OpenBSD, :NetBSD, :Darwin, :Apple))
+    @eval is_unix()    = $(is_linux() || is_bsd())
+    @eval is_windows() = $(KERNEL == :NT)
+    export is_apple, is_linux, is_bsd, is_unix, is_windows
+else
+    const KERNEL = Sys.KERNEL
+end
+
```
Comment (c42f) [inline src/DeclarativePackages.jl:24]: I think all this stuff comes for free in Compat.jl, is there a reason we can't just use that?

Comment (r-chris) [reply src/DeclarativePackages.jl:24]: I think we initially wanted to drop support for anything before 0.5 - not sure if there would be a disadvantage in using Compat.jl

Comment (JoshChristie) [reply src/DeclarativePackages.jl:24]: The problem is that the environment variable `JULIA_PKGDIR` changes the `Pkg.dir()` and so having `using Compat` will look in that directory where it won't exists.

**Pull request merged**